catalyst-gateway/bin/src/state/mod.rs

```diff
@@ -49,8 +50,50 @@ impl State {
         Ok(state)
     }
 
-    // pub(crate) async fn event_db(&self) -> Option<Arc<dyn EventDbQueries>> {
-    //
-    //
-    // }
+    /// Get the reference to the database connection pool for `EventDB`.
+    pub(crate) fn event_db(&self) -> Option<Arc<dyn EventDbQueries>> {
+        let guard = self._schema_version_status_lock();
+        match *guard {
+            SchemaVersionStatus::Ok => Some(self.event_db.clone()),
+            SchemaVersionStatus::Mismatch => None,
+        }
+    }
+
+    /// Check the DB schema version matches the one expected by the service.
+    pub(crate) async fn schema_version_check(&self) -> Result<i32, Error> {
+        Ok(self.event_db.schema_version_check().await?)
+    }
+
+    /// Compare the `State`'s inner value with a given `&SchemaVersionStatus`, returns
+    /// `bool`.
+    pub(crate) fn is_schema_version_status(&self, svs: &SchemaVersionStatus) -> bool {
+        let guard = self._schema_version_status_lock();
+        &*guard == svs
+    }
+
+    /// Set the state's `SchemaVersionStatus`.
+    pub(crate) fn set_schema_version_status(&self, svs: SchemaVersionStatus) {
+        let mut guard = self._schema_version_status_lock();
+        tracing::debug!(
+            status = format!("{:?}", svs),
+            "db schema version status was set"
+        );
+        *guard = svs;
+    }
+
+    /// Get the `MutexGuard<SchemaVersionStatus>` from inner the variable.
+    ///
+    /// Handle poisoned mutex by recovering the guard, and tracing the error.
+    fn _schema_version_status_lock(&self) -> MutexGuard<SchemaVersionStatus> {
```

Comment:
Why underscore as a first character ?

Reply:
It's just an old convention for naming low-level functions that aren't really meant to be in the API. I have no personal attachment to keeping it, just did so out of habit.